karapace/backup/backends/v3/backend.py

```diff
@@ -88,6 +89,40 @@ def read(self, path: Path, topic_name: str) -> Iterator[Instruction]:
                 data_file=data_file,
             )
 
+    # Not part of common interface, because it exposes DataFile which is
+    # specific to V3.
+    def verify_files(self, path: Path) -> Generator[tuple[bool, DataFile], None, None]:
+        metadata = self.read_metadata(path)
+
+        for data_file in metadata.data_files:
+            checksum = _get_checksum_implementation(metadata.checksum_algorithm)()
+            with (path.parent / data_file.filename).open("rb") as buffer:
+                while True:
+                    chunk = buffer.read(io.DEFAULT_BUFFER_SIZE)
+                    if not chunk:
+                        break
+                    checksum.update(chunk)
+            matches = checksum.digest() == data_file.checksum
+            yield matches, data_file
+
+    # Not part of common interface, because it exposes DataFile which is
+    # specific to V3.
+    def verify_records(self, path: Path) -> Generator[tuple[bool, DataFile], None, None]:
+        metadata = self.read_metadata(path)
+
+        for data_file in metadata.data_files:
+            try:
+                for _ in self._read_data_file(
+                    path=(path.parent / data_file.filename),
+                    metadata=metadata,
+                    data_file=data_file,
+                ):
+                    pass
+            except DecodeError:
+                yield False, data_file
```

Comment:
The exception has some extra information, offset, record num etc. This information now gets omitted, would it be any value to print out?

Reply:
Fair point, it probably makes sense to be ambitious and have this from the get-go. 👍

Reply:
The API layer now yields results as `VerifyFailure | VerifySuccess`, where `VerifyFailure` has an optional exception that is formatted in the command.